### PR TITLE
STY: use pytest.raises context manager (io.sql)

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref #24332

### CODE REVIEW

File: pandas/tests/io/test_sql.py [hunk 1]
```diff
@@ -2663,23 +2669,17 @@ def clean_up(test_table_to_drop):
             self.drop_table(test_table_to_drop)
 
         # test if invalid value for if_exists raises appropriate error
-        pytest.raises(ValueError,
-                      sql.to_sql,
-                      frame=df_if_exists_1,
-                      con=self.conn,
-                      name=table_name,
-                      if_exists='notvalidvalue')
+        with pytest.raises(ValueError, match="<insert message here>"):
```
Comment (simonjayhawkins) [inline pandas/tests/io/test_sql.py]: dummy message for unconditionally skipped test. xref #20536

Comment (gfyoung) [reply pandas/tests/io/test_sql.py]: Yep, that makes sense.

**Review by gfyoung — APPROVED** [review]
cc @jreback

**Review by WillAyd — APPROVED** [review]

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**